test/libraries/SafeMigration.spec.ts

```diff
@@ -0,0 +1,365 @@
+                    COMPATIBILITY_FALLBACK_HANDLER = (await (await user1.sendTransaction({ data: to?.safeL2DeploymentData })).wait())
+                        ?.contractAddress;
+
+                    migration = (await hre.ethers.deployContract("SafeMigration", [
+                        SAFE_SINGLETON_ADDRESS,
+                        SAFE_SINGLETON_L2_ADDRESS,
+                        COMPATIBILITY_FALLBACK_HANDLER,
+                    ])) as unknown as SafeMigration;
+                }
+
+                const singletonAddress = (await (await user1.sendTransaction({ data: from.safeDeploymentData })).wait())?.contractAddress;
+                const singletonL2Address = (await (await user1.sendTransaction({ data: from.safeL2DeploymentData })).wait())
+                    ?.contractAddress;
+
+                if (!singletonAddress || !singletonL2Address) {
+                    throw new Error("Could not deploy safe or safeL2");
+                }
+                const singleton = await getSafeSingletonAt(singletonAddress);
+                const singletonL2 = await getSafeSingletonAt(singletonL2Address);
+                console.log("ddssd");
+
+                return {
+                    signers,
+                    safe: await getSafeWithSingleton(singleton, [user1.address]),
+                    safeL2: await getSafeWithSingleton(singletonL2, [user1.address]),
+                    migration,
+                };
+            });
+
+            describe("constructor", () => {
+                it("reverts when Safe singleton is not a contract", async () => {
+                    await setupTests();
+
+                    await expect(
+                        hre.ethers.deployContract("SafeMigration", [
+                            ethers.ZeroAddress,
+                            SAFE_SINGLETON_L2_ADDRESS,
+                            COMPATIBILITY_FALLBACK_HANDLER,
+                        ]),
+                    ).to.be.revertedWith("Safe Singleton is not deployed");
+                });
+
+                it("reverts when SafeL2 singleton is not a contract", async () => {
+                    await setupTests();
+
+                    await expect(
+                        hre.ethers.deployContract("SafeMigration", [
+                            SAFE_SINGLETON_ADDRESS,
+                            ethers.ZeroAddress,
+                            COMPATIBILITY_FALLBACK_HANDLER,
+                        ]),
+                    ).to.be.revertedWith("Safe Singleton (L2) is not deployed");
+                });
+
+                it("reverts when fallback handler is not a contract", async () => {
+                    await setupTests();
+
+                    await expect(
+                        hre.ethers.deployContract("SafeMigration", [SAFE_SINGLETON_ADDRESS, SAFE_SINGLETON_L2_ADDRESS, ethers.ZeroAddress]),
+                    ).to.be.revertedWith("fallback handler is not deployed");
+                });
+            });
```

Comment:
uber-nit: I don't think this needs to be part of the test suite. In particular, I don't see value in repeating constructor tests for each version.

Reply:
Agree on not repeating it, but I won't recommend removing it completely either. Maybe no matter the version, the address passed should always have a code in it should be a check.

Reply:
https://github.com/safe-global/safe-smart-account/pull/793/commits/15bdca03f4590d44c926d9ffd779adea6eac5f27

This uses only latest versions of contract because constructor only checks for presence of code. So, using latest contracts and addresses should be fine.